GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs

```diff
@@ -8,12 +8,49 @@
 
 namespace GitUI.CommandsDialogs.BrowseDialog.DashboardControl
 {
+    public interface IRepositoryFilter
+    {
+        void Filter(IList<Repository> repositories);
+    }
+
+    /// <summary>
+    /// Case insensitive text search based on the repository path
+    /// </summary>
+    public class RepositoryTextFilter : IRepositoryFilter
+    {
+        private readonly string _pattern;
+
+        public RepositoryTextFilter(string pattern)
+        {
+            _pattern = pattern.ToLower();
+        }
+
+        public void Filter(IList<Repository> repositories)
+        {
+            if (_pattern.Trim().Length == 0)
+            {
+                return;
+            }
+
+            for (var i = repositories.Count - 1; i >= 0; i--)
+            {
+                var repo = repositories[i];
+                if (repo.Path.ToLower().Contains(_pattern))
```

Comment:
For me, it's better to lower none of the strings (even the pattern above) and instead use `StringComparison.OrdinalIgnoreCase`

https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings

Reply:
The comparison should probably use the machine's locale. Not everyone uses Latin characters. If suggest using StringComparison.CurrentCultureIgnoreCase.